src/cmds/qsub.c

```diff
@@ -6218,8 +6226,10 @@ main(int argc, char **argv, char **envp) /* qsub */
 	basic_envlist = job_env_basic();
 	if (basic_envlist == NULL)
 		exit_qsub(3);
-	if (V_opt)
-		qsub_envlist = env_array_to_varlist(envp);
+	
+	/* qsub_envlist might be needed if -V option is set in default flags.
+	 * It will be unset if not needed then. */
+	qsub_envlist = env_array_to_varlist(envp);
```

Comment:
We need to call the function env_array_to_varlist() only when desired i.e. when V_opt is set either while calling qsub or through default_qsub_arguments.
 In other words before calling the function set_job_env() make sure that V_opt(This should be set only when qsub is called through -V option or default_qsub_arguments contains -V option) is set or not. If set then only call env_array_to_varlist to populate qsub_envlist and then pass it to set_job_env()